praw/models/reddit/mixins/__init__.py

```diff
@@ -134,10 +139,54 @@ def remove(self, spam=False):
            # remove a submission
            submission = reddit.submission(id='5or86n')
            submission.mod.remove()
+           # remove a submission with a removal reason
+           reason = reddit.subreddit.mod.removal_reasons["110ni21zo23ql"]
+           submission = reddit.submission(id="5or86n")
+           submission.mod.remove(reason_id=reason.id)
 
         """
         data = {"id": self.thing.fullname, "spam": bool(spam)}
         self.thing._reddit.post(API_PATH["remove"], data=data)
+        if any([reason_id, mod_note]):
+            self._add_removal_reason(mod_note, reason_id)
+
+    def _add_removal_reason(self, mod_note="", reason_id=None):
+        """Add a removal reason for a Comment or Submission.
+
+        :param mod_note: A message for the other moderators.
+        :param reason_id: The removal reason ID.
+
+        It is necessary to first call :meth:`~.remove` on the
+        :class:`~.Comment` or :class:`~.Submission`.
+
+        If reason_id is not specified, mod_note cannot be blank.
+
+        Example usage:
```

Comment:
Could we remove just the example from the docstring please? That way we don't encourage people to use a private method.